rfc-134-govuk-wide-session-cookie-and-login.md

```diff
@@ -0,0 +1,359 @@
+# A GOV.UK-wide session cookie & login
+
+## Summary
+
+We propose introducing a new cookie, `__Host-govuk_account_session`,
+which will be an essential cookie, but only set when a user signs in
+to use personalised parts of GOV.UK (currently just the Transition
+Checker).
+
+Similarly to how our A/B tests work we will manage this cookie at the
+Fastly layer, in Varnish Configuration Language (VCL), and use custom
+request headers to pass the cookie value to our apps. We will also use
+custom response headers to set a new cookie value.
+
+We will create a new app to provide the internal account-supporting
+API, and extend frontend to handle the login and logout process.
+
+## Problem
+
+The GOV.UK Account team have launched an experiment on the Transition
+Checker, allowing users to sign up to save their
+results. [finder-frontend sets a session cookie][ff-session]—an
+encrypted cookie—containing the user ID and OAuth tokens used to
+update the data we hold on them.
+
+This works fine for one app, but there are problems with this approach
+when we try to scale to more than one app.
+
+[ff-session]: https://github.com/alphagov/finder-frontend/blob/5b6101e8eb027028a102ab96bcef6ff752b9c558/app/controllers/sessions_controller.rb#L23-L28
+
+### We need to pass cookies to our apps
+
+Our Nginx configuration blocks most cookies, [which we disabled for
```

Comment:
@richardTowers Our of interest, will we have this sort of thing (a proxy removing undesirable headers, both inbound and outbound) in the replatformed world?

Reply:
I've jotted it down on my board of things the current platform does, but it's not in the plan at the moment. I'm interested in how valuable we think the feature is.

Reply:
I suspect the main value we get out of it is not needing to confirm whether any of our apps try to set session cookies by default when they shouldn't do.